src/config.c

```diff
@@ -578,6 +621,27 @@ int parse_args (int argc, char **argv, GromitData *data)
                wrong_arg = TRUE;
              }
          }
+       else if (strcmp (arg, "-H") == 0 ||
+                strcmp (arg, "--history-steps") == 0)
+       {
+         if (i+1 < argc && atoi (argv[i+1]) > 0)
+         {
+           data->undo_steps = atoi (argv[i+1]);
+           data->undobuffer = calloc(data->undo_steps, sizeof(cairo_surface_t*));
+
+           for (int i = 0; i < data->undo_steps; i++)
```

Comment:
Same here, this should be done in one single place (where it is done already - main?)

Reply:
It is done inside `setup_main_app`. Should a new function at main.c be created that allocates the necessary resources and is called by everyone else?

Reply:
IIRC the flow is more or less as this: 

* main fires up, https://github.com/bk138/gromit-mpx/blob/master/src/main.c#L1017
* it then decides whether to act as the "main app" or the "client app", https://github.com/bk138/gromit-mpx/blob/master/src/main.c#L1078
* in setup_main_app, you'd do allocations based on the value of data->undo_steps, https://github.com/bk138/gromit-mpx/blob/master/src/main.c#L583 
  * **only there**, the other places you just get the value of data->undo_steps